### PR TITLE
feat: added an enlint rule to enforce no-type-import

### LINKED ISSUE #1735: Add an eslint rule to enforce no-type-import and add the why in the docs.
## Scope & Context

We just removed all type imports from the codebase in #1679, now it would be helpful for people to enforce this with an eslint rule and a section in the documentation.

There's already an existing rule : https://typescript-eslint.io/rules/consistent-type-imports/

And you can find the discussion in the associated [ticket](https://github.com/twentyhq/twenty/issues/1679) and [PR](https://github.com/twentyhq/twenty/pull/1716) to have some material for writing in the documentation in https://docs.twenty.com/developer/frontend/style-guide

### PR DESCRIPTION
**Pull request by aman44444**
Closes #1735 

Description:
-Added an Enlint rule to enforce not-type-import.
-Updated in Docs.

### CODE REVIEW

File: docs/docs/developer/frontend/style-guide.mdx [hunk 1]
```diff
@@ -259,3 +259,32 @@ const StyledButton = styled.button`
...
+import { Meta, StoryObj } from '@storybook/react';
+```
+
+### Why No-Type Imports?
+
+- **Consistency**: By avoiding type imports, our codebase remains consistent in its module import style. We use a single approach for both type and value imports.
+
+- **Readability**: No-type imports improve code readability by making it clear when you're importing values or types. This reduces ambiguity and makes it easier to understand the purpose of imported symbols.
+
+- **Maintainability**: Codebase maintainability is enhanced because developers can quickly identify and locate type-only imports when reviewing or modifying code.
```
Comment (lucasbordeau) [inline docs/docs/developer/frontend/style-guide.mdx:284]: Could you please add a reference to the documentation which explains it is meant only for rare edge cases : https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-8.html

**Review by lucasbordeau — APPROVED** [review]

**Comment by lucasbordeau** [comment]
@aman44444 Could you open a PR to remove the type import in the storybook test stories at it crashes the CI ?

**Comment by aman44444** [comment]
@lucasbordeau sure

**Comment by aman44444** [comment]
@lucasbordeau could you please add Hacktoberfest label or Hacktoberfest accepted? it will be helpful for me

**Comment by lucasbordeau** [comment]
@charlesBochet @FelixMalfait  What are your thoughts ?

**Comment by FelixMalfait** [comment]
@aman44444 thanks a lot for your contribution! 
The `hacktoberfest-accepted label` isn't required since we have the hacktoberfest tag on the repo, all merged PR should be taken into account!

**Pull request merged**